tests/v2/actions/projects/projects.go

```diff
@@ -0,0 +1,102 @@
+package projects
+
+import (
+	"strings"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/kubeapi/namespaces"
+	projectsapi "github.com/rancher/shepherd/extensions/kubeapi/projects"
+	"github.com/rancher/shepherd/extensions/projects"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+// NewProjectTemplate is a constructor that creates the project template
```

Comment:
can we move this to shepherd extensions please?

Reply:
Created a PR to move this to shepherd: https://github.com/rancher/shepherd/pull/220